core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -250,13 +250,39 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
         String groupId,
         Map<TopicIdPartition, List<ShareAcknowledgementBatch>> acknowledgeTopics
     ) {
-        log.trace("Acknowledge request for topicIdPartitions: {} with groupId: {}",
+        log.debug("Acknowledge request for topicIdPartitions: {} with groupId: {}",
             acknowledgeTopics.keySet(), groupId);
+        Map<TopicIdPartition, CompletableFuture<Errors>> futures = new HashMap<>();
+        acknowledgeTopics.forEach((topicIdPartition, acknowledgePartitionBatches) -> {
+            SharePartition sharePartition = partitionCacheMap.get(sharePartitionKey(groupId, topicIdPartition));
+            if (sharePartition != null) {
+                synchronized (sharePartition) {
```

Comment:
Why do we require this lock?

Reply:
We did earlier, but with the use of locks in the Share Partition, we do not anymore. I will make this change